### PR TITLE
Remove deprecated ioutil package and update modules

### PR DESCRIPTION
**Pull request by chimanjain**
# Description

* Remove deprecated `ioutil` package
* Update go modules
* Remove unwanted build stage in `podman-build`
* Minor formatting error fixes

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/885 |
| https://github.com/dell/csm/issues/916 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Run Sanity Test
 
Screenshot 2023-08-14 111849
Screenshot 2023-08-14 111909

### CODE REVIEW

File: common/constants/consts.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ const (
 	// cluster when no size is given, expressed in bytes
 	DefaultVolumeSizeInBytes = 3 * BytesInGiB
 
-	// BytesInGiB is the number of bytes in a gibibyte
+	// BytesInGiB is the number of bytes in a gibabyte
```
Comment (nitesh3108) [inline common/constants/consts.go:36]: Spell-check *gigabyte*

Comment (chimanjain) [reply common/constants/consts.go]: fixed

**Review by nitesh3108 — APPROVED** [review]
LGTM

**Review by HarishH-DELL — APPROVED** [review]
LGTM

**Pull request merged**